From 82d7a770e06dee1d0b59be8d3e973abe1e7ef199 Mon Sep 17 00:00:00 2001
From: Craig Buchek <craig@boochtek.com>
Date: Tue, 29 Jul 2008 23:53:30 -0500
Subject: [PATCH] Fix bug #99 - quoting issues in sqlite3 adapter.

Add SQLite3Adapter#quote_table_name; use single quotes instead of double quotes.
---
 .../connection_adapters/sqlite3_adapter.rb         |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb b/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb
index cc9c465..8cb03f4 100644
--- a/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb
@@ -24,6 +24,10 @@ module ActiveRecord
 
   module ConnectionAdapters #:nodoc:
     class SQLite3Adapter < SQLiteAdapter # :nodoc:
+      def quote_table_name(table_name)
+        %Q('#{table_name}')
+      end
+
       def table_structure(table_name)
         returning structure = @connection.table_info(quote_table_name(table_name)) do
           raise(ActiveRecord::StatementInvalid, "Could not find table '#{table_name}'") if structure.empty?
-- 
1.5.2.5

